### PR TITLE
Added files for mod kernel

### PR DESCRIPTION
**Pull request by carrycooldude**
>I made a P.R for mod kernel and solves this issue [#5110](https://github.com/tensorflow/tfjs/issues/5110)

To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5317)

### CODE REVIEW

**Review by pyu10055 — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/5317)* status: 0 of 1 approvals obtained (waiting on @carrycooldude)

___
*[tfjs-backend-wasm/src/cc/kernels/Mod.cc, line 45 at r1](https://reviewable.io/reviews/tensorflow/tfjs/5317#-MeRZdmYAg7a5pBmFSLH:-MeRZdmYAg7a5pBmFSLI:b-8ps9wv) ([raw file](https://github.com/tensorflow/tfjs/blob/ca6d80c9419f2938e3e43b5a20efb57ceef65b59/tfjs-backend-wasm/src/cc/kernels/Mod.cc#L45)):*
> ```cpp
>     case DType::int32:
>       binary_i32(a_id, b_id, out_id, [](int a, int b) {
>         return static_cast<int32_t>(fmod(static_cast<float>(a) , b));
> ```

can you use a % b for int32 dtype?

**Comment by raffizulvian** [comment]
Build failed due to formatting/lint error, not related to xnnpack (google/XNNPACK#1613) I think. You can see it on the build log.

**Comment by carrycooldude** [comment]
> Build failed due to formatting/lint error, not related to xnnpack ([google/XNNPACK#1613](https://github.com/google/XNNPACK/pull/1613)) I think. You can see it on the build log.

Can you tell me how to resolve lint error

**Comment by raffizulvian** [comment]
You can just look at the build log or run ```yarn run lint``` locally inside tfjs-backend-wasm directory. This command will output the log contain information of which line causes the error and what action is needed to fix the error.

**Comment by carrycooldude** [comment]
Screenshot from 2021-07-28 17-54-07
Screenshot from 2021-07-28 17-54-01
Still I think so there is some logic error after I resolve lint error

**Comment by carrycooldude** [comment]
```
binary_f32(a_id, b_id, out_id,
                 [ ](float a, float b) {
                   float mod = fmod(a, b);
                   if ((a < 0 && b < 0) || (a >= 0 && b >= 0)) {
                     return mod;
                   } else {
                     return fmod((mod+b)  , b);
                   }
                  }); 
```

This is the logic which I am applying for that

**Comment by vladmandic** [comment]
any updates on this one?

**Comment by carrycooldude** [comment]
> any updates on this one?

I also have a conversation with xnnpack repo moderator too for this mod kernel is not supporting 
https://github.com/google/XNNPACK/pull/1613

**Comment by kinsei0916** [comment]
Can you implement mod kernel in TFJS first? I guess it would take long to implement in XNNPACK.

File: tfjs-backend-wasm/src/cc/kernels/Mod.cc [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+
+    return mod;
+                  });
+      break;
+    case DType::int32:
+      binary_i32(a_id, b_id, out_id, [](int a, int b) {
+        return a >= 0 ? a % b : ( b - abs ( a%b ) ) % b;
+      });
+      break;
```
Comment (kinsei0916) [inline tfjs-backend-wasm/src/cc/kernels/Mod.cc:77]: This algorithm seems different from you commented: https://github.com/tensorflow/tfjs/pull/5317#issuecomment-888399451
I don't know what's happened here, but [this commit](https://github.com/tensorflow/tfjs/pull/5317/commits/32dacee7c7dcd63fb47406b10e3587b42158cfee) broke the code. Can you revert the commit and run lint again?

**Comment by kinsei0916** [comment]
Even after fixing that, the test might still fail with error related to broadcasting.
If so, can you add `'broadcasts 2x1 Tensor2D and 2x2 Tensor2D'` into exclude list in setup_test.ts?

File: tfjs-backend-wasm/src/cc/kernels/Mod.cc (hunk 1 above)
Comment (carrycooldude) [reply tfjs-backend-wasm/src/cc/kernels/Mod.cc:77]: Sure

**Comment by chunnienc** [comment]
WASM implementation has been added through #4236 and #7670 . Please let me know if you run into other issues with this op, and thanks for your contributions!

**Pull request closed without merging**